shop-the-look/api/v1/endpoints/image.py

```diff
@@ -0,0 +1,76 @@
+import os
+import base64
+import requests
+import imghdr
+from fastapi import APIRouter, UploadFile, File, HTTPException
+from api.config import settings
+from api import deps
+
+router = APIRouter()
+
+@router.post("/search/image")
+async def query_image(file: UploadFile = File(...)):
+    try:
+        # Check file format
+        contents = await file.read()
+        file_format = imghdr.what(None, h=contents)
+        
+        # Vertex AI Multimodal Embedding Model only supports the following image formats
+        if file_format not in ['bmp', 'gif', 'jpeg', 'png', 'jpg']:
+            raise HTTPException(status_code=400, detail="We only support BMP, GIF, JPG, and PNG for images. Please upload a valid image file.")
+        
+        # Encode image to base64
+        base64_encoded_image = base64.b64encode(contents).decode('utf-8')
+        
+        # Get the access token
+        access_token = settings.get_access_token()
+        
+        # Prepare the API request
+        url = f"https://{settings.location}-aiplatform.googleapis.com/v1/projects/{settings.project_id}/locations/{settings.location}/publishers/google/models/multimodalembedding@001:predict"
```

Comment:
NIT: extract this into a utility function to format the URL properly by taking its component parts.

Reply:
Done